src/components/Board.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect } from "react";
+import CreateTaskInput from "./CreateTaskInput";
+import List from "./List";
+import CustomCard from "./CustomCard";
+import "./styles.css";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  createColumnAction,
+  createTaskAction,
+  deleteColumnAction,
+  deleteTaskAction,
+  getAppDataAction,
+  updateTaskColumnIdAction,
+} from "../redux/actions/inputAction";
+import { Button, Input } from "semantic-ui-react";
+import "semantic-ui-css/semantic.min.css";
+
+export interface Column {
+  id: number;
+  title: string;
+  tasks: Task[];
+}
+
+export interface Task {
+  title: string;
+  description: string;
+  columnId: number;
+  id: number;
+}
+
+export default function Board(): JSX.Element {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getAppDataAction());
+  }, [dispatch]);
+
+  const appData = useSelector(({ app }: { app: Column[] }) => app);
+
+  const GetMoveButtonsForTask = (taskId: number) => {
+    return (
+      <div className="move_buttons">
+        {appData.map((column) => (
+          <Button
+            size="mini"
+            key={column.id}
+            onClick={() =>
+              dispatch(updateTaskColumnIdAction(taskId, column.id))
+            }
+          >
+            {column.title}
+          </Button>
+        ))}
+      </div>
+    );
+  };
+
+  const AddColumnButton = (): JSX.Element => {
+    return (
+      <div>
+        <Input
+          size="mini"
+          placeholder="new column name"
+          onKeyPress={(event: { key: string; target: { value: string } }) => {
+            if (event.key === "Enter") {
+              dispatch(createColumnAction(event.target.value));
+            }
+          }}
+        ></Input>
+      </div>
+    );
+  };
+
+  const GetDeleteColumnButton = (columnId: number) => {
+    return (
+      <div>
+        <Button
+          size="mini"
+          negative
+          className="delete_column_btn"
+          onClick={() => dispatch(deleteColumnAction(columnId))}
+        >
+          X
+        </Button>
+      </div>
+    );
+  };
+
+  return (
+    <div className="board">
+      {appData.map((column) => {
+        return (
+          <List
+            key={column.id}
+            deleteColumnButton={GetDeleteColumnButton(column.id)}
+            title={column.title}
+          >
+            {column.tasks.map((task: Task) => {
+              return (
```

Comment:
`task: Task` is not necessary 
since `appData` is already fully typed

Reply:
fixed